frontends/web/src/components/guide/guide.tsx

```diff
@@ -14,86 +14,46 @@
  * limitations under the License.
  */
 
-import { ReactNode, useEffect } from 'react';
+import { ReactNode, useContext, useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
-import { share } from '../../decorators/share';
-import { Store } from '../../decorators/store';
-import { translate, TranslateProps } from '../../decorators/translate';
-import { setConfig } from '../../utils/config';
 import { apiGet } from '../../utils/request';
 import A from '../anchor/anchor';
 import { CloseXWhite } from '../icon';
+import AppContext from '../../contexts/AppContext';
 import style from './guide.module.css';
 
-export type TSharedProps = {
-    shown: boolean;
-    // eslint-disable-next-line react/no-unused-prop-types
-    activeSidebar: boolean;
-    // eslint-disable-next-line react/no-unused-prop-types
-    sidebarStatus: string;
-    // eslint-disable-next-line react/no-unused-prop-types
-    guideExists: boolean;
-}
-
-export type TProps = TSharedProps & {
+export type TProps = {
     children?: ReactNode;
 }
 
-export const store = new Store<TSharedProps>({
-  shown: false,
-  activeSidebar: false,
-  sidebarStatus: '',
-  guideExists: false,
-});
-
-// if apiGet() is invoked immediately this can error due to cyclic dependencies
-// request.js:64 Uncaught TypeError: Cannot read properties of undefined
-// (reading 'runningInQtWebEngine')
-// TODO: this should probably be in a context
-setTimeout(() => {
-  apiGet('config').then(({ frontend }) => {
-    if (frontend && frontend.guideShown !== undefined) {
-      store.setState({ shown: frontend.guideShown });
-    } else {
-      store.setState({ shown: true });
-    }
-  });
-}, 0);
-
-function setGuideShown(shown: boolean) {
-  store.setState({ shown });
-  setConfig({ frontend: { guideShown: shown } });
-}
-
-export function toggle() {
-  setGuideShown(!store.state.shown);
-}
-
-export function show() {
-  setGuideShown(true);
-}
-
-export function hide() {
-  setGuideShown(false);
-}
-
-const Guide = ({ shown, children }: TProps) => {
+const Guide = ({ children }: TProps) => {
+  const { setGuideShown, guideShown: shown, toggleGuide, setGuideExists } = useContext(AppContext);
 
   useEffect(() => {
-    store.setState({ guideExists: true });
+    setGuideExists(true);
     return () => {
-      store.setState({ guideExists: false });
+      setGuideExists(false);
     };
-  }, []);
+  }, [setGuideExists]);
+
+  useEffect(() => {
+    apiGet('config').then(({ frontend }) => {
+      if (frontend && frontend.guideShown !== undefined) {
+        setGuideShown(frontend.guideShown);
+      } else {
+        setGuideShown(true);
+      }
+    });
+  }, [setGuideShown]);
```

Comment:
since you are here, please change to `import { getConfig } from '../../utils/config';`

Reply:
this changes the behavior, before config was only read once on application start. With this PR it gets the config now each time on component mount (if I read correctly).

If we go back to the old behavior, should this go into AppProvider?

Reply:
Ah yeah.

> If we go back to the old behavior, should this go into AppProvider?

Absolutely right 👍